src/main/Port.Adapter/UI/Views/Blazor.Common/TreeView.razor

```diff
@@ -316,6 +319,80 @@
             "";
     }
 
+    private async Task<string> Hover(TreeNeuronViewModel child, bool recursive = false, string id = "")
+    {
+        try
+        {
+            return await ProcessHoverOrUnhover(child, recursive, id, true);
+        }
+        catch (Exception ex)
+        {
+            return ex.Message;
+        }
+    }
+
+    private async Task<string> Unhover(TreeNeuronViewModel child, bool recursive = false, string id = "")
+    {
+        try
+        {
+            return await ProcessHoverOrUnhover(child, recursive, id, false);
+        }
+        catch (Exception ex)
+        {
+            return ex.Message;
+        }
+    }
+
+    private async Task<string> ProcessHoverOrUnhover(TreeNeuronViewModel child, bool recursive, string id, bool isHover)
+    {
+        if (!this.Children.Contains(child))
+        {
+            var newChild = this.Children.FirstOrDefault(x => x.Children.Contains(child));
+            if (newChild != null)
+            {
+                child = newChild;
+            }
+        }
+
+        await HandleNodeHighlight(child, recursive, id, isHover);
+
+        return "";
+    }
+
+    private async Task HandleNodeHighlight(TreeNeuronViewModel node, bool recursive, string id, bool isHover)
+    {
+        // Highlight or unhighlight the current node
+        if (!recursive)
+        {
+            await JSRuntime.InvokeVoidAsync("hover", "g", node.Neuron.Id, isHover, "highlightNode");
+        }
+
+        // Process children if the node is expanded and has children
+        if (node.ExpansionState == ExpansionState.Expanded && node.Children?.Any() == true)
+        {
+            foreach (var childNode in node.Children)
+            {
+                // Skip processing if the child node matches the given ID
+                if (childNode.Neuron.Id == id || HoverCore(childNode, id))
+                    continue;
+
+                // Highlight or unhighlight the child node
+                await JSRuntime.InvokeVoidAsync("hover", "g", childNode.Neuron.Id, isHover, "highlightChildNode");
+
+                // Recursively process children of the child node if expanded
+                if (childNode.ExpansionState == ExpansionState.Expanded && childNode.Children?.Any() == true)
+                {
+                    await HandleNodeHighlight(childNode, true, id, isHover);
+                }
+            }
+        }
+    }
+
+    private bool HoverCore(TreeNeuronViewModel childNode, string id)
```

Comment:
When called from line 376, isn't the first condition already duplicated? childNode.Neuron.Id == id

If so, is this function still needed? If not, let's transfer the second condition to 376 to simplify.